ob1k-http/src/main/java/com/outbrain/ob1k/http/ning/NingHttpTypedStreamHandler.java

```diff
@@ -38,44 +40,45 @@ public NingHttpTypedStreamHandler(final long responseMaxSize, final Observer<Typ
   }
 
   @Override
-  public STATE onBodyPartReceived(final HttpResponseBodyPart bodyPart) throws Exception {
-
+  public State onBodyPartReceived(final HttpResponseBodyPart bodyPart) throws Exception {
     if (responseMaxSize > 0) {
-      responseSizesAggregated += bodyPart.length();
-      if (responseSizesAggregated > responseMaxSize) {
+      if (responseSizesAggregated.addAndGet(bodyPart.length()) > responseMaxSize) {
         onThrowable(new RuntimeException("Response size is bigger than the limit: " + responseMaxSize));
-        return STATE.ABORT;
+        return State.ABORT;
       }
     }
 
-    final com.ning.http.client.Response ningResponse = new NettyResponse(status, headers, singletonList(bodyPart));
-    final TypedResponse<T> response = new NingResponse<>(ningResponse, type, marshallingStrategy);
+    final org.asynchttpclient.Response ningResponse = new NettyResponse(status, headers, singletonList(bodyPart));
+    final TypedResponse<T> response = new AsyncHttpResponse<>(ningResponse, type, marshallingStrategy);
 
+    if (bodyPart.isLast()) {
```

Comment:
Had to add this to prevent marshal exceptions

Reply:
https://github.com/outbrain/ob1k/blob/ahc_upgrade/ob1k-http/src/main/java/com/outbrain/ob1k/http/providers/ning/NingHttpStreamHandler.java#L33

did it almost two years ago :)
check that branch out, it basically contains 99% of the things you need.